SMLHelper/Options/Attributes/MenuAttribute.cs

```diff
@@ -0,0 +1,200 @@
+    public sealed class MenuAttribute : Attribute
+    {
+        /// <summary>
+        /// Specifies after which events the config file should be saved to disk automatically.
+        /// </summary>
+        /// <remarks>
+        /// This enumeration has a <see cref="FlagsAttribute"/> that allows a bitwise combination of its member values.
+        /// </remarks>
+        [Flags]
+        public enum SaveEvents : byte
+        {
+            /// <summary>
+            /// Never automatically save.
+            /// </summary>
+            None = 0,
+            /// <summary>
+            /// Save whenever any value is changed.
+            /// </summary>
+            ChangeValue = 1,
+            /// <summary>
+            /// Save when the player saves the game.
+            /// </summary>
+            SaveGame = 2,
+            /// <summary>
+            /// Save when the player quits the game.
+            /// </summary>
+            QuitGame = 4
+        }
+
+        /// <summary>
+        /// Specifies after which events the config file should be loaded from disk automatically.
+        /// </summary>
+        /// <remarks>
+        /// This enumeration has a <see cref="FlagsAttribute"/> that allows a bitwise combination of its member values.
+        /// </remarks>
+        [Flags]
+        public enum LoadEvents : byte
+        {
+            /// <summary>
+            /// Never automatically load.
+            /// </summary>
+            None = 0,
+            /// <summary>
+            /// Load when the menu is registered to SMLHelper via <see cref="Handlers.OptionsPanelHandler.RegisterModOptions{T}"/>.
+            /// </summary>
+            /// <remarks>
+            /// In normal usage, this option is equivalent to loading when the game is launched.
+            /// </remarks>
+            MenuRegistered = 1,
+            /// <summary>
+            /// Load when the menu is opened by the player.
+            /// </summary>
+            /// <remarks>
+            /// Useful for allowing players to edit their config files manually without restarting the game for their changes to take effect.
+            /// </remarks>
+            MenuOpened = 2
+        }
+
+        /// <summary>
+        /// Specifies which members of the <see cref="ConfigFile"/> will be parsed and added to the menu.
+        /// </summary>
+        public enum Members
+        {
+            /// <summary>
+            /// Only <see langword="public"/> members attributed with a <see cref="ModOptionAttribute"/> derivative such as 
+            /// <see cref="SliderAttribute"/>, <see cref="ChoiceAttribute"/> etc. will be processed.
+            /// </summary>
+            OptIn,
+            /// <summary>
+            /// All <see langword="public"/> members will be processed, and where they are not attributed, a default attribute
+            /// will be generated for them.
+            /// </summary>
+            OptOut
```

Comment:
Consider alternatives to `OptOut` since it doesn't exactly describe what this is for.

`IncludeAll` or `Implicit` or maybe something around there?

Reply:
How about `OptIn` = `Explicit` & `OptOut` = `Implicit`?

Reply:
But my reasons for `OptOut` is that you can still opt out - it will imply that this is an attribute unless you opt-out of it by using the `IgnoreMemberAttribute`. Similar in design to Newtonsoft.Json's `OptIn` vs `OptOut` + `JsonIgnore`.

Reply:
With `OptIn` & `OptOut`, I do feel like you are right in that if you're looking at it, you have to remind yourself which one results in which behaviour. `Explicit` & `Implicit` I think is likely easier to grasp. So I'm on-board with the change.

Reply:
I like `Explicit`/`Implicit` 👍

Reply:
81d410e :)